src/main/java/org/opentripplanner/routing/edgetype/StreetEdgeTraversal.java

```diff
@@ -0,0 +1,282 @@
+package org.opentripplanner.routing.edgetype;
+
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.routing.core.StateEditor;
+import org.opentripplanner.routing.core.TraverseMode;
+import org.opentripplanner.routing.graph.Edge;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.routing.util.ElevationUtils;
+import org.opentripplanner.routing.vertextype.IntersectionVertex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Created by kathrynkillebrew on 1/23/15.
+ */
+public class StreetEdgeTraversal {
+
+    private static Logger LOG = LoggerFactory.getLogger(StreetEdgeTraversal.class);
+
+    private static final double GREENWAY_SAFETY_FACTOR = 0.1;
+
+    /** return a StateEditor rather than a State so that we can make parking/mode switch modifications for kiss-and-ride. */
+    public static StateEditor doTraverse(StreetEdge edge, State s0, RoutingRequest options, TraverseMode traverseMode) {
```

Comment:
I may do that in working on the weighting logic.  It is strange having a class with just one static method and no state; things could probably be shuffled around a bit more, though it's not clear to me yet what be the most sensible approach here.